extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -219,12 +227,29 @@ export class Bind {
     /** @const @private {!Deferred} */
     this.addMacrosDeferred_ = new Deferred();
 
-    /** @private {Promise} */
+    /** @private {?Promise} */
     this.setStatePromise_ = null;
 
     /** @private @const {!../../../src/utils/signals.Signals} */
     this.signals_ = new Signals();
 
+    /**
+     * A queue of history replace/push to adhere to browser rate-limits.
+     * @private @const {!Array<{
+     *   op: !HistoryOp,
+     *   data: (!JsonObject|undefined),
+     *   onPop: (!Function|undefined)
+     * }>}
+     */
+    this.historyQueue_ = [];
+
+    /** @private @const {!Function} */
+    this.debouncedFlushHistoryQueue_ = debounce(
+      this.win_,
+      () => this.flushHistoryQueue_(),
+      1000
```

Comment:
Depending on how quickly these calls are coming in, we may never replace the history. Eg, if a `slidechange` happens every .5s, this debounce will never trigger (it's waiting for 1s to pass since the last call before invoking).